src/framework/ui/view/uitheme.cpp

```diff
@@ -354,10 +352,7 @@ void UiTheme::setupUiFonts()
 
         font->setPixelSize(size);
         font->setFamily(QString::fromStdString(family));
-
-        if (weight == QFont::DemiBold) {
```

Comment:
I implemented it in https://github.com/musescore/MuseScore/pull/7097 for fixing bold fonts on Windows and macOS
@DmitryArefiev FYI need to check